builtin/mainmenu/content/pkgmgr.lua

```diff
@@ -197,11 +199,21 @@ function pkgmgr.get_texture_packs()
 		return a.title:lower() < b.title:lower()
 	end)
 
-	return retval
+	pkgmgr.texture_packs = retval
+end
+
+--------------------------------------------------------------------------------
+function pkgmgr.get_texture_packs()
+	if not pkgmgr.texture_packs then
+		pkgmgr.reload_texture_packs()
+	end
```

Comment:
As far as I can see, `pkgmgr.load_all()` is called before every use of `pkgmgr.get_texture_packs()` and this is unnecessary.

And if this is unnecessary, why is there even a getter function?

- `pkgmgr.games` / `pkgmgr.reload_games()`
- `pkgmgr.global_mods:get_list()` / `pkgmgr.reload_global_mods()`
- but: `pkgmgr.get_texture_packs()` / `pkgmgr.reload_texture_packs()`